docs/paper/dev/api/event-api/chat-event.md

```diff
@@ -0,0 +1,138 @@
+---
+slug: /dev/chat-events
+description: An outline on the AsyncChatEvent and how to handle it 
+---
+
+# Chat Events
+
+The Chat Event has evolved a few times over the years. 
+This guide will explain how to properly use the new `AsyncChatEvent` and its `ChatRenderer`.
+The `AsyncChatEvent` is an improved version of the old `AsyncPlayerChatEvent` that allows you to render chat messages individually for each player.
+
+:::note
+
+#### AsyncChatEvent vs ChatEvent
+
+The key difference between the `AsyncChatEvent` and the `ChatEvent` is that the `AsyncChatEvent` is fired asynchronously 
+which means that it does not block the main thread and delays the chat message.
+Be aware that using the Bukkit API in the event might cause errors. 
+If you need to use the Bukkit API, use the `ChatEvent` instead.
+Alternatively you can use the [`BukkitScheduler`](../scheduler.md).
+
+:::
+
+## Understanding the renderer
+
+Before we can start using the new chat event, we need to understand how the new renderer works.
+The renderer is Paper's way of allowing plugins to modify the chat message before it is sent to the player.
+This is done by using the `ChatRenderer` interface with its `render` method. Previously, this was done by using the `AsyncPlayerChatEvent` with its `setFormat` method.
+
+```java title="ChatRenderer#render"
+public @NotNull Component render(@NotNull Player source, @NotNull Component sourceDisplayName, @NotNull Component message, @NotNull Audience viewer) {
+    // ...
+}
+```
+
+- The `render` method is called when a chat message is sent to the player.
+- The `source` parameter is the player that sent the message.
+- The `sourceDisplayName` parameter is the display name of the player that sent the message.
+- The `message` parameter is the message that was sent.
+- The `viewer` parameter is the player that is receiving the message.
```

Comment:
am i right in saying this is often not a singular player but instead can be a collection of players?

Reply:
Well in terms of this method it's a singular audience.